src/cpp/src/llm_pipeline_stateful.cpp

```diff
@@ -40,7 +40,12 @@ StatefulLLMPipeline::StatefulLLMPipeline(
     const ov::genai::GenerationConfig& generation_config)
     : LLMPipelineImplBase(tokenizer, generation_config), m_sampler(m_tokenizer) {
     utils::apply_slice_before_matmul_transformation(model);
-    m_kv_history_manager.kv_cache_seq_length_axis = ov::genai::utils::get_seq_len_axis(model);
+
+    if (device == "NPU")
```

Comment:
```suggestion
    if (device.find("NPU") != std::string::npos)
```
because device can be `NPU.1` in case of NPU with several tails.